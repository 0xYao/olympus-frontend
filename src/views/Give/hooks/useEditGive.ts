import { t } from "@lingui/macro";
import { ContractReceipt, ethers } from "ethers";
import { useMutation, useQueryClient } from "react-query";
import { useDispatch } from "react-redux";
import { GIVE_ADDRESSES, SOHM_ADDRESSES } from "src/constants/addresses";
import { useWeb3Context } from "src/hooks";
import { balanceQueryKey } from "src/hooks/useBalance";
import { useDynamicGiveContract } from "src/hooks/useContract";
import { donationInfoQueryKey } from "src/hooks/useGiveInfo";
import { useTestableNetworks } from "src/hooks/useTestableNetworks";
import { error as createErrorToast, info as createInfoToast } from "src/slices/MessagesSlice";

import { GiveData } from "../Interfaces";

/**
 * @notice Increases the value of an active donation
 * @returns ContractReceipt for the deposit
 */
export const useIncreaseGive = () => {
  const dispatch = useDispatch();
  const client = useQueryClient();
  const { address } = useWeb3Context();
  const networks = useTestableNetworks();
  const contract = useDynamicGiveContract(GIVE_ADDRESSES, true);

  // Mutation to interact with the YieldDirector contract
  return useMutation<ContractReceipt, Error, GiveData>(
    // Pass in an object with an amount and a recipient parameter
    async ({ amount: amount_, recipient: recipient_ }) => {
      // Validate inputs
      if (parseFloat(amount_) <= 0) throw new Error(t`An increase Give amount must be positive`);

      // Confirm that the user is on a chain where YieldDirector exists
      if (!contract) throw new Error(t`Please switch to the Ethereum network to donate yield`);

      // Create transaction to deposit passed amount to the passed recipient
      const transaction = await contract.deposit(ethers.utils.parseUnits(amount_, "gwei"), recipient_);
      return transaction.wait();
    },
    {
      onError: error => {
        dispatch(createErrorToast(error.message));
      },
      onSuccess: async () => {
        // Refetch sOHM balance and donation info
        const keysToRefetch = [
          balanceQueryKey(address, SOHM_ADDRESSES, networks.MAINNET),
          donationInfoQueryKey(address, networks.MAINNET),
        ];

        const promises = keysToRefetch.map(key => client.refetchQueries(key, { active: true }));
        await Promise.all(promises);

        dispatch(createInfoToast(t`Successfully deposited sOHM`));
      },
    },
  );
};

/**
 * @notice Decreases the value of an active donation
 * @returns ContractReceipt for the deposit
 */
export const useDecreaseGive = () => {
  const dispatch = useDispatch();
  const client = useQueryClient();
  const { address } = useWeb3Context();
  const networks = useTestableNetworks();
  const contract = useDynamicGiveContract(GIVE_ADDRESSES, true);

  // Mutation to interact with the YieldDirector contract
  return useMutation<ContractReceipt, Error, GiveData>(
    // Pass in an object with an amount and a recipient parameter
    async ({ amount: amount_, recipient: recipient_ }) => {
      // Validate inputs
      if (parseFloat(amount_) <= 0) throw new Error(t`A decrease Give amount must be positive`);

      // Confirm that the user is on a chain where YieldDirector exists
      if (!contract) throw new Error(t`Please switch to the Ethereum network to donate yield`);

      // Create transaction to withdraw passed amount from the passed recipient
      const transaction = await contract.withdraw(ethers.utils.parseUnits(amount_, "gwei"), recipient_);
      return transaction.wait();
    },
    {
      onError: error => {
        dispatch(createErrorToast(error.message));
      },
      onSuccess: async () => {
        // Refetch sOHM balance and donation info
        const keysToRefetch = [
          balanceQueryKey(address, SOHM_ADDRESSES, networks.MAINNET),
          donationInfoQueryKey(address, networks.MAINNET),
        ];

        const promises = keysToRefetch.map(key => client.refetchQueries(key, { active: true }));
        await Promise.all(promises);

        dispatch(createInfoToast(t`Successfully deposited sOHM`));
      },
    },
  );
};
