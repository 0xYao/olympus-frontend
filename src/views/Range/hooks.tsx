import { gql, request } from "graphql-request";
import { useQuery } from "react-query";
//import { RANGE_CONTRACT, RANGE_PRICE_CONTRACT } from "src/constants/contracts";
// import { NetworkId } from "src/networkDetails";

export const OHMPriceHistory = (address: string) => {
  const graphURL = "https://api.thegraph.com/subgraphs/name/uniswap/uniswap-v3";
  const {
    data = [],
    isFetched,
    isLoading,
  } = useQuery(["OHMPriceHistory", address], async () => {
    const data = await request(
      graphURL,
      gql`
        {
          tokenDayDatas(
            where: { token: "${address}" }
            orderBy: date
            orderDirection: desc
            first: 7
          ) {
            priceUSD
            date
          }
        }
      `,
    );
    return data.tokenDayDatas;
  });

  return { data, isFetched, isLoading };
};

/**
 *
 * @param address
 * @returns Returns the capacity of the Operator at the given address
 */
export const Capacity = (address: string) => {
  //const contract = RANGE_CONTRACT.getEthersContract(NetworkId.MAINNET);
  const { data, isFetched, isLoading } = useQuery(["Capacity", address], async () => {
    //const capacty = await contract.lastMarketCapacity(false);
    //TODO: REMOVE STUB RESPONSE
    return 10000;
  });
  return { data, isFetched, isLoading };
};

/**
 * @param address
 * @returns Returns the upper and lower range of the Operator at the given address
 */
export const RangeBoundaries = (address: string) => {
  // const contract = RANGE_CONTRACT.getEthersContract(NetworkId.MAINNET);
  const {
    data = { high: 0, low: 0, cushion: 0, wall: 0 },
    isFetched,
    isLoading,
  } = useQuery(["Capacity", address], async () => {
    //TODO: REMOVE STUB RESPONSE
    return { low: 2000, high: 2500, cushion: 1000, wall: 10000 };
  });
  return { data, isFetched, isLoading };
};

/**
 * @param address
 * @returns Returns the current price of the Operator at the given address
 */
export const OperatorPrice = (address: string) => {
  //const contract = RANGE_PRICE_CONTRACT.getEthersContract(NetworkId.MAINNET);
  const {
    data = 0,
    isFetched,
    isLoading,
  } = useQuery(["OperatorPrice", address], async () => {
    //contract.getCurrentPrice();
    //TODO: REMOVE STUB RESPONSE
    return "$15.50";
  });
  return { data, isFetched, isLoading };
};
