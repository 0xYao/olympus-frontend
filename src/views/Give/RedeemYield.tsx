import { t, Trans } from "@lingui/macro";
import { Grid, Typography } from "@material-ui/core";
import { useTheme } from "@material-ui/core/styles";
import useMediaQuery from "@material-ui/core/useMediaQuery";
import { Skeleton } from "@material-ui/lab";
import { DataRow, PrimaryButton } from "@olympusdao/component-library";
import { BigNumber } from "bignumber.js";
import { useEffect, useState } from "react";
import { useDispatch, useSelector } from "react-redux";
import { useLocation } from "react-router-dom";
import { GiveBox as Box } from "src/components/GiveProject/GiveBox";
import { NetworkId } from "src/constants";
import { Environment } from "src/helpers/environment/Environment/Environment";
import { useWeb3Context } from "src/hooks/web3Context";
import { loadAccountDetails } from "src/slices/AccountSlice";
import { isPendingTxn, txnButtonText } from "src/slices/PendingTxnsSlice";

import { Project } from "../../components/GiveProject/project.type";
import { redeemBalance, redeemMockBalance } from "../../slices/RedeemThunk";
import { DonationInfoState } from "./Interfaces";
import data from "./projects.json";
import { RedeemCancelCallback, RedeemYieldModal } from "./RedeemYieldModal";

// Consistent with staking page
const DECIMAL_PLACES = 4;

export default function RedeemYield() {
  const location = useLocation();
  const dispatch = useDispatch();
  const { provider, address, connected, networkId } = useWeb3Context();
  const [isRedeemYieldModalOpen, setIsRedeemYieldModalOpen] = useState(false);
  const theme = useTheme();
  const isSmallScreen = useMediaQuery(theme.breakpoints.down("sm"));
  const { projects } = data;
  const projectMap = new Map(projects.map(i => [i.wallet, i] as [string, Project]));

  const isAppLoading = useSelector((state: DonationInfoState) => state.app.loading);

  const redeemableBalance = useSelector((state: DonationInfoState) => {
    return networkId === NetworkId.TESTNET_RINKEBY && Environment.isMockSohmEnabled(location.search)
      ? state.account.mockRedeeming && state.account.mockRedeeming.sohmRedeemable
      : state.account.redeeming && state.account.redeeming.sohmRedeemable;
  });

  const recipientInfo = useSelector((state: DonationInfoState) => {
    return state.account.redeeming && state.account.redeeming.recipientInfo;
  });

  const stakingRebase = useSelector((state: DonationInfoState) => {
    return state.app.stakingRebase;
  });

  const fiveDayRate = useSelector((state: DonationInfoState) => {
    return state.app.fiveDayRate;
  });

  const pendingTransactions = useSelector((state: DonationInfoState) => {
    return state.pendingTransactions;
  });

  const redeemableBalanceNumber: BigNumber = new BigNumber(redeemableBalance);

  const totalDeposit = new BigNumber(recipientInfo && recipientInfo.totalDebt ? recipientInfo.totalDebt : 0);

  const stakingRebasePercentage = new BigNumber(stakingRebase ? stakingRebase : 0).multipliedBy(100);
  const nextRewardValue = new BigNumber(stakingRebase ? stakingRebase : 0).multipliedBy(totalDeposit);

  const fiveDayRateValue = new BigNumber(fiveDayRate ? fiveDayRate : 0).multipliedBy(100);

  const isProject = projectMap.get(address);

  const isRecipientInfoLoading = recipientInfo.totalDebt == "";

  // this useEffect fires on state change from above. It will ALWAYS fire AFTER
  useEffect(() => {
    // don't load ANY details until wallet is Checked
    if (connected) {
      loadAccountDetails({ networkID: networkId, provider, address });
    }
  }, [connected]);

  /**
   * Get project sOHM yield goal and return as a BigNumber
   *
   * @param address
   * @returns
   */
  const getRecipientGoal = (address: string): BigNumber => {
    const project = projectMap.get(address);
    if (project) return new BigNumber(project.depositGoal);

    return new BigNumber(0);
  };

  /**
   * Checks that the current user can redeem some quantity of sOHM
   *
   * @returns
   */
  const canRedeem = () => {
    if (!address) return false;

    if (isRecipientInfoLoading) return false;

    if (isPendingTxn(pendingTransactions, "redeeming")) return false;

    if (redeemableBalanceNumber.isEqualTo(0))
      // If the available amount is 0
      return false;

    return true;
  };

  const handleRedeemButtonClick = () => {
    setIsRedeemYieldModalOpen(true);
  };

  const handleRedeemYieldModalSubmit = async () => {
    if (networkId === NetworkId.TESTNET_RINKEBY && Environment.isMockSohmEnabled(location.search)) {
      await dispatch(redeemMockBalance({ address, provider, networkID: networkId, eventSource: "Redeem" }));
    } else {
      await dispatch(redeemBalance({ address, provider, networkID: networkId, eventSource: "Redeem" }));
    }
    setIsRedeemYieldModalOpen(false);
  };

  const handleRedeemYieldModalCancel: RedeemCancelCallback = () => {
    setIsRedeemYieldModalOpen(false);
  };

  return (
    <Grid container spacing={2}>
      <Grid item xs={12}>
        <Typography variant="h3" align="center">
          {isRecipientInfoLoading ? <Skeleton /> : redeemableBalanceNumber.toFormat(DECIMAL_PLACES)} sOHM
        </Typography>
        <Typography variant="body1" align="center" className="subtext">
          Redeemable Yield
        </Typography>
      </Grid>
      <Grid item xs={12}>
        <Grid container>
          <Grid item xs />
          <Grid item xs={12} sm={6}>
            <PrimaryButton onClick={() => handleRedeemButtonClick()} disabled={!canRedeem()} fullWidth>
              {txnButtonText(pendingTransactions, "redeeming", t`Redeem Yield`)}
            </PrimaryButton>
          </Grid>
          <Grid item xs />
        </Grid>
      </Grid>
      {isProject ? (
        <Grid item xs={12}>
          <Grid container spacing={1}>
            <Grid item xs={4}>
              <Box>
                <Typography variant="h5" align="center">
                  {getRecipientGoal(address).toFormat(DECIMAL_PLACES)}
                </Typography>
                <Typography variant="body1" align="center" className="subtext">
                  <Trans>sOHM Goal</Trans>
                </Typography>
              </Box>
            </Grid>
            <Grid item xs={4}>
              <Box>
                <Typography variant="h5" align="center">
                  {totalDeposit.toFormat(DECIMAL_PLACES)}
                </Typography>
                <Typography variant="body1" align="center" className="subtext">
                  {isSmallScreen ? t`Total Donated` : t`Total sOHM Donated`}
                </Typography>
              </Box>
            </Grid>
            <Grid item xs={4}>
              <Box>
                <Typography variant="h5" align="center">
                  {totalDeposit.multipliedBy(100).div(getRecipientGoal(address)).toFormat(DECIMAL_PLACES)}%
                </Typography>
                <Typography variant="body1" align="center" className="subtext">
                  <Trans>of sOHM Goal</Trans>
                </Typography>
              </Box>
            </Grid>
          </Grid>
        </Grid>
      ) : (
        <></>
      )}
      <Grid item xs={12}>
        <Box>
          <DataRow
            title={t`Deposited sOHM`}
            // Exact number
            balance={`${totalDeposit.toFormat()} ${t`sOHM`}`}
            isLoading={isRecipientInfoLoading}
          />
          <DataRow
            title={t`Redeemable Amount`}
            // Exact number
            balance={`${redeemableBalanceNumber.toFormat()} ${t`sOHM`}`}
            isLoading={isRecipientInfoLoading}
          />
          <DataRow
            title={t`Next Reward Amount`}
            // Exact number
            balance={`${nextRewardValue.toFormat(DECIMAL_PLACES)} ${t`sOHM`}`}
            isLoading={isAppLoading}
          />
          <DataRow
            title={t`Next Reward Yield`}
            balance={`${stakingRebasePercentage.toFormat(DECIMAL_PLACES)}%`}
            isLoading={isAppLoading}
          />
          <DataRow
            title={t`ROI (5-Day Rate)`}
            balance={`${fiveDayRateValue.toFormat(DECIMAL_PLACES)}%`}
            isLoading={isAppLoading}
          />
        </Box>
      </Grid>
      <Grid item>
        <RedeemYieldModal
          isModalOpen={isRedeemYieldModalOpen}
          callbackFunc={handleRedeemYieldModalSubmit}
          cancelFunc={handleRedeemYieldModalCancel}
          deposit={totalDeposit}
          redeemableBalance={redeemableBalanceNumber}
        />
      </Grid>
    </Grid>
  );
}
