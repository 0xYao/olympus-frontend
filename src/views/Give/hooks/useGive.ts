import { t } from "@lingui/macro";
import { ContractReceipt, ethers } from "ethers";
import { useMutation, useQueryClient } from "react-query";
import { useDispatch } from "react-redux";
import { GIVE_ADDRESSES, SOHM_ADDRESSES } from "src/constants/addresses";
import { useWeb3Context } from "src/hooks";
import { balanceQueryKey } from "src/hooks/useBalance";
import { useDynamicGiveContract } from "src/hooks/useContract";
import { donationInfoQueryKey } from "src/hooks/useGiveInfo";
import { useTestableNetworks } from "src/hooks/useTestableNetworks";
import { error as createErrorToast, info as createInfoToast } from "src/slices/MessagesSlice";

import { GiveData } from "../Interfaces";

/**
 * @notice Creates a new deposit
 * @returns ContractReceipt for the deposit
 */
export const useGive = () => {
  const dispatch = useDispatch();
  const client = useQueryClient();
  const { address } = useWeb3Context();
  const networks = useTestableNetworks();
  const contract = useDynamicGiveContract(GIVE_ADDRESSES, true);

  // Mutation to interact with the YieldDirector contract
  return useMutation<ContractReceipt, Error, GiveData>(
    // Pass in an object with an amount and a recipient parameter
    async ({ amount: amount_, recipient: recipient_ }) => {
      // Validate inputs
      if (parseFloat(amount_) <= 0) throw new Error(t`A give amount must be positive`);

      // Confirm that the user is on a chain where YieldDirector exists
      if (!contract) throw new Error(t`Please switch to the Ethereum network to donate yield`);

      // Create transaction to deposit passed amount to the passed recipient
      const transaction = await contract.deposit(ethers.utils.parseUnits(amount_, "gwei"), recipient_);
      return transaction.wait();
    },
    {
      onError: error => {
        console.error(error.message);
        dispatch(createErrorToast(error.message));
      },
      onSuccess: async () => {
        const keysToRefetch = [
          balanceQueryKey(address, SOHM_ADDRESSES, networks.MAINNET),
          donationInfoQueryKey(address, networks.MAINNET),
        ];

        const promises = keysToRefetch.map(key => client.refetchQueries(key, { active: true }));
        await Promise.all(promises);

        dispatch(createInfoToast(t`Successfully deposited sOHM`));
      },
    },
  );
};
