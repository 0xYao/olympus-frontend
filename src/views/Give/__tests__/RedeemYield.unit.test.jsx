import { fireEvent } from "@testing-library/dom";
import * as GiveInfo from "src/hooks/useGiveInfo";
import * as useWeb3Context from "src/hooks/web3Context";
import { mockWeb3Context } from "src/testHelpers";

import { act, render, screen } from "../../../testUtils";
import RedeemYield from "../RedeemYield";

let context;
beforeEach(() => {
  context = jest.spyOn(useWeb3Context, "useWeb3Context");
});

afterEach(() => {
  //jest.restoreAllMocks();
  jest.clearAllMocks();
});

describe("Redeem Yield", () => {
  it("should render Redeem Yield Screen", async () => {
    context.mockReturnValue(mockWeb3Context);
    let container;
    await act(async () => {
      ({ container } = render(<RedeemYield />));
    });
    expect(container).toMatchSnapshot();
  });

  it("should have disabled redeem button when there are pending transaction(s)", async () => {
    context.mockReturnValue(mockWeb3Context);

    let container;
    await act(async () => {
      ({ container } = render(<RedeemYield />)); //eslint-disable-line
    });

    fireEvent.click(screen.getByText("Redeem Yield").closest("button"));
    expect(screen.getByText("Redeem Yield").closest("button")).toHaveAttribute("disabled");
  });

  it("should show redeemable balance as 100 sOHM", async () => {
    context.mockReturnValue(mockWeb3Context);
    const redeemable = jest.spyOn(GiveInfo, "useRedeemableBalance");
    redeemable.mockReturnValue({ data: "100" });

    let container;
    await act(async () => {
      ({ container } = render(<RedeemYield />)); //eslint-disable-line
    });
    expect(container).toMatchSnapshot();
    expect(screen.getByTestId("redeemable-balance")).toHaveTextContent("100 sOHM");
  });

  it("should show extra content if project wallet", async () => {
    context.mockReturnValue({ ...mockWeb3Context, address: "0xd3B4a9604c78DDA8692d85Dc15802BA12Fb82b6c" });
    const redeemable = jest.spyOn(GiveInfo, "useRedeemableBalance");
    redeemable.mockReturnValue({ data: "100" });

    let container;
    await act(async () => {
      ({ container } = render(<RedeemYield />)); //eslint-disable-line
    });
    expect(screen.getByText("sOHM Goal")).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
});
