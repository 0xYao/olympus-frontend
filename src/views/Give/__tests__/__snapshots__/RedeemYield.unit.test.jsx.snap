// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Redeem Yield should render Redeem Yield Screen 1`] = `
<div>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2"
  >
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
    >
      <h3
        class="MuiTypography-root MuiTypography-h3 MuiTypography-alignCenter"
        data-testid="redeemable-balance"
      >
        <span
          class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse"
        />
         sOHM
      </h3>
      <p
        class="MuiTypography-root subtext MuiTypography-body1 MuiTypography-alignCenter"
      >
        Redeemable Yield
      </p>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
    >
      <div
        class="MuiGrid-root MuiGrid-container"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-true"
        />
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-root-1 makeStyles-root-2 undefined MuiButton-containedPrimary MuiButton-disableElevation Mui-disabled MuiButton-fullWidth Mui-disabled"
            disabled=""
            tabindex="-1"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              Redeem Yield
            </span>
          </button>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-true"
        />
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
    >
      <div
        class="MuiBox-root MuiBox-root-3"
      >
        <div
          class="makeStyles-root-4 data-row"
        >
          <div
            class="MuiBox-root MuiBox-root-6"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 MuiTypography-colorPrimary"
            >
              Deposited sOHM
            </p>
          </div>
          <p
            class="MuiTypography-root MuiTypography-body1 MuiTypography-colorPrimary"
            style="text-align: right;"
          >
            <span
              class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse"
              style="width: 80px;"
            />
          </p>
        </div>
        <div
          class="makeStyles-root-4 data-row"
        >
          <div
            class="MuiBox-root MuiBox-root-7"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 MuiTypography-colorPrimary"
            >
              Redeemable Amount
            </p>
          </div>
          <p
            class="MuiTypography-root MuiTypography-body1 MuiTypography-colorPrimary"
            style="text-align: right;"
          >
            <span
              class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse"
              style="width: 80px;"
            />
          </p>
        </div>
        <div
          class="makeStyles-root-4 data-row"
        >
          <div
            class="MuiBox-root MuiBox-root-8"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 MuiTypography-colorPrimary"
            >
              Next Reward Amount
            </p>
          </div>
          <p
            class="MuiTypography-root MuiTypography-body1 MuiTypography-colorPrimary"
            style="text-align: right;"
          >
            0 sOHM
          </p>
        </div>
        <div
          class="makeStyles-root-4 data-row"
        >
          <div
            class="MuiBox-root MuiBox-root-9"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 MuiTypography-colorPrimary"
            >
              Next Reward Yield
            </p>
          </div>
          <p
            class="MuiTypography-root MuiTypography-body1 MuiTypography-colorPrimary"
            style="text-align: right;"
          >
            0%
          </p>
        </div>
        <div
          class="makeStyles-root-4 data-row"
        >
          <div
            class="MuiBox-root MuiBox-root-10"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 MuiTypography-colorPrimary"
            >
              ROI (5-Day Rate)
            </p>
          </div>
          <p
            class="MuiTypography-root MuiTypography-body1 MuiTypography-colorPrimary"
            style="text-align: right;"
          >
            0%
          </p>
        </div>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item"
    />
  </div>
</div>
`;

exports[`Redeem Yield should show extra content if project wallet 1`] = `
<div>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2"
  >
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
    >
      <h3
        class="MuiTypography-root MuiTypography-h3 MuiTypography-alignCenter"
        data-testid="redeemable-balance"
      >
        <span
          class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse"
        />
         sOHM
      </h3>
      <p
        class="MuiTypography-root subtext MuiTypography-body1 MuiTypography-alignCenter"
      >
        Redeemable Yield
      </p>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
    >
      <div
        class="MuiGrid-root MuiGrid-container"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-true"
        />
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-root-45 makeStyles-root-46 undefined MuiButton-containedPrimary MuiButton-disableElevation Mui-disabled MuiButton-fullWidth Mui-disabled"
            disabled=""
            tabindex="-1"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              Redeem Yield
            </span>
          </button>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-true"
        />
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
    >
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4"
        >
          <div
            class="MuiBox-root MuiBox-root-47"
          >
            <h5
              class="MuiTypography-root MuiTypography-h5 MuiTypography-alignCenter"
              data-testid="project-goal"
            >
              200
            </h5>
            <p
              class="MuiTypography-root subtext MuiTypography-body1 MuiTypography-alignCenter"
            >
              sOHM Goal
            </p>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4"
        >
          <div
            class="MuiBox-root MuiBox-root-48"
          >
            <h5
              class="MuiTypography-root MuiTypography-h5 MuiTypography-alignCenter"
              data-testid="project-deposit"
            >
              0
            </h5>
            <p
              class="MuiTypography-root subtext MuiTypography-body1 MuiTypography-alignCenter"
            >
              Total sOHM Donated
            </p>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4"
        >
          <div
            class="MuiBox-root MuiBox-root-49"
          >
            <h5
              class="MuiTypography-root MuiTypography-h5 MuiTypography-alignCenter"
              data-testid="project-goal-achievement"
            >
              0
              %
            </h5>
            <p
              class="MuiTypography-root subtext MuiTypography-body1 MuiTypography-alignCenter"
            >
              of sOHM Goal
            </p>
          </div>
        </div>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
    >
      <div
        class="MuiBox-root MuiBox-root-50"
      >
        <div
          class="makeStyles-root-51 data-row"
        >
          <div
            class="MuiBox-root MuiBox-root-53"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 MuiTypography-colorPrimary"
            >
              Deposited sOHM
            </p>
          </div>
          <p
            class="MuiTypography-root MuiTypography-body1 MuiTypography-colorPrimary"
            style="text-align: right;"
          >
            <span
              class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse"
              style="width: 80px;"
            />
          </p>
        </div>
        <div
          class="makeStyles-root-51 data-row"
        >
          <div
            class="MuiBox-root MuiBox-root-54"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 MuiTypography-colorPrimary"
            >
              Redeemable Amount
            </p>
          </div>
          <p
            class="MuiTypography-root MuiTypography-body1 MuiTypography-colorPrimary"
            style="text-align: right;"
          >
            <span
              class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse"
              style="width: 80px;"
            />
          </p>
        </div>
        <div
          class="makeStyles-root-51 data-row"
        >
          <div
            class="MuiBox-root MuiBox-root-55"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 MuiTypography-colorPrimary"
            >
              Next Reward Amount
            </p>
          </div>
          <p
            class="MuiTypography-root MuiTypography-body1 MuiTypography-colorPrimary"
            style="text-align: right;"
          >
            0 sOHM
          </p>
        </div>
        <div
          class="makeStyles-root-51 data-row"
        >
          <div
            class="MuiBox-root MuiBox-root-56"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 MuiTypography-colorPrimary"
            >
              Next Reward Yield
            </p>
          </div>
          <p
            class="MuiTypography-root MuiTypography-body1 MuiTypography-colorPrimary"
            style="text-align: right;"
          >
            0%
          </p>
        </div>
        <div
          class="makeStyles-root-51 data-row"
        >
          <div
            class="MuiBox-root MuiBox-root-57"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 MuiTypography-colorPrimary"
            >
              ROI (5-Day Rate)
            </p>
          </div>
          <p
            class="MuiTypography-root MuiTypography-body1 MuiTypography-colorPrimary"
            style="text-align: right;"
          >
            0%
          </p>
        </div>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item"
    />
  </div>
</div>
`;

exports[`Redeem Yield should show redeemable balance as 100 sOHM 1`] = `
<div>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2"
  >
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
    >
      <h3
        class="MuiTypography-root MuiTypography-h3 MuiTypography-alignCenter"
        data-testid="redeemable-balance"
      >
        100
         sOHM
      </h3>
      <p
        class="MuiTypography-root subtext MuiTypography-body1 MuiTypography-alignCenter"
      >
        Redeemable Yield
      </p>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
    >
      <div
        class="MuiGrid-root MuiGrid-container"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-true"
        />
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-root-27 makeStyles-root-28 undefined MuiButton-containedPrimary MuiButton-disableElevation MuiButton-fullWidth"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              Redeem Yield
            </span>
          </button>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-true"
        />
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
    >
      <div
        class="MuiBox-root MuiBox-root-29"
      >
        <div
          class="makeStyles-root-30 data-row"
        >
          <div
            class="MuiBox-root MuiBox-root-40"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 MuiTypography-colorPrimary"
            >
              Deposited sOHM
            </p>
          </div>
          <p
            class="MuiTypography-root MuiTypography-body1 MuiTypography-colorPrimary"
            style="text-align: right;"
          >
            0 sOHM
          </p>
        </div>
        <div
          class="makeStyles-root-30 data-row"
        >
          <div
            class="MuiBox-root MuiBox-root-41"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 MuiTypography-colorPrimary"
            >
              Redeemable Amount
            </p>
          </div>
          <p
            class="MuiTypography-root MuiTypography-body1 MuiTypography-colorPrimary"
            style="text-align: right;"
          >
            100 sOHM
          </p>
        </div>
        <div
          class="makeStyles-root-30 data-row"
        >
          <div
            class="MuiBox-root MuiBox-root-42"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 MuiTypography-colorPrimary"
            >
              Next Reward Amount
            </p>
          </div>
          <p
            class="MuiTypography-root MuiTypography-body1 MuiTypography-colorPrimary"
            style="text-align: right;"
          >
            0 sOHM
          </p>
        </div>
        <div
          class="makeStyles-root-30 data-row"
        >
          <div
            class="MuiBox-root MuiBox-root-43"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 MuiTypography-colorPrimary"
            >
              Next Reward Yield
            </p>
          </div>
          <p
            class="MuiTypography-root MuiTypography-body1 MuiTypography-colorPrimary"
            style="text-align: right;"
          >
            0%
          </p>
        </div>
        <div
          class="makeStyles-root-30 data-row"
        >
          <div
            class="MuiBox-root MuiBox-root-44"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 MuiTypography-colorPrimary"
            >
              ROI (5-Day Rate)
            </p>
          </div>
          <p
            class="MuiTypography-root MuiTypography-body1 MuiTypography-colorPrimary"
            style="text-align: right;"
          >
            0%
          </p>
        </div>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item"
    />
  </div>
</div>
`;
